src/GraphQL/Caching/IDocumentCache.cs

```diff
@@ -8,8 +9,13 @@ namespace GraphQL.Caching
     public interface IDocumentCache
     {
         /// <summary>
-        /// Gets or sets a document in the cache. Must be thread-safe. Returns <see langword="null"/> if no entry is found.
+        /// Gets a document in the cache. Must be thread-safe. Returns <see langword="null"/> if no entry is found.
         /// </summary>
-        Document? this[string query] { get; set; }
+        ValueTask<Document?> GetAsync(string query);
+
+        /// <summary>
+        /// Sets a document in the cache. Must be thread-safe.
+        /// </summary>
+        Task SetAsync(string query, Document? value);
```

Comment:
`Document?` or `Document` ?

Reply:
Also change `Task` to `ValueTask` to align with one from `GetAsync`. No performance reasons here, just unifying.

Reply:
Also this PR should change documentation/migration guide for v5 accordingly.

Reply:
Can be a separate PR but it's better to handle documentation changes here as well.

Reply:
> `Document?` or `Document` ?

https://github.com/graphql-dotnet/graphql-dotnet/pull/2758/commits/a3301366655be6be93c1b14c5aef25e37ba2b289

Reply:
> Can be a separate PR but it's better to handle documentation changes here as well.

Need help on docs changes because I'm not familiar with graphql-dotnet. 🍰

Reply:
If you can just add a note under the breaking changes section of this page:
- https://github.com/graphql-dotnet/graphql-dotnet/blob/develop/docs2/site/docs/migrations/migration5.md

I’d suggest saying that the indexer property of IDocumentCache has been replaced with GetAsync, SetAsync and RemoveAsync methods or similar.

Reply:
You can make the change within the same PR. It should be pretty easy to do.

Reply:
(Usually we will make another pass at the documentation before release so it doesn’t have to be perfect, but it helps remind us what changes were made and what needs to be documented. )

Reply:
> You can make the change within the same PR. It should be pretty easy to do.

https://github.com/gdlcf88/graphql-dotnet/commit/32e2d31a7c2c905d92fff88ef3734011ef6999b8